quartz_solar_forecast/inverters/enphase.py

```diff
@@ -1,47 +1,154 @@
+    print("Please copy and paste the full redirect URL here:")
+    redirect_url = input()
+    # Extract the authorization code from the redirect URL
+    code = redirect_url.split("?code=")[1]
+    return code
+
 
 def get_enphase_access_token():
     """
-    Obtain an access token for the Enphase API using the Client Credentials Grant flow.
+    Obtain an access token for the Enphase API using the Authorization Code Grant flow.
+    :param None
+    :return: Access Token
     """
-    url = "https://api.enphaseenergy.com/oauth/token"
+        
+    client_id = os.getenv('ENPHASE_CLIENT_ID')
+    client_secret = os.getenv('ENPHASE_CLIENT_SECRET')
+
+    auth_url = get_enphase_auth_url()
+    auth_code = get_enphase_authorization_code(auth_url)
+
+    # Combine the client ID and secret with a colon separator
+    credentials = f"{client_id}:{client_secret}"
+
+    # Encode the credentials as bytes
+    credentials_bytes = credentials.encode("utf-8")
+
+    # Base64 encode the bytes
+    encoded_credentials = base64.b64encode(credentials_bytes)
+
+    # Convert the encoded bytes to a string
+    encoded_credentials_str = encoded_credentials.decode("utf-8")
+    # print("Base64 encoded credentials:", encoded_credentials_str)
+
+    conn = http.client.HTTPSConnection("api.enphaseenergy.com")
+    payload = ""
     headers = {
-        "Content-Type": "application/x-www-form-urlencoded",
-        "Authorization": f"Basic {(ENPHASE_CLIENT_ID + ':' + ENPHASE_CLIENT_SECRET).encode().decode('utf-8')}",
-    }
-    data = {
-        "grant_type": "client_credentials",
-        "scope": "read"
+        "Authorization": f"Basic {encoded_credentials_str}"
     }
+    conn.request(
+        "POST",
+        f"/oauth/token?grant_type=authorization_code&redirect_uri=https://api.enphaseenergy.com/oauth/redirect_uri&code={auth_code}",
+        payload,
+        headers,
+    )
+    res = conn.getresponse()
+    data = res.read()
+
+    # Decode the data read from the response
+    decoded_data = data.decode("utf-8")
+    # print("UTF-8 DECODED DATA:\n", data.decode("utf-8"))
+
+    # Convert the decoded data into JSON format
+    data_json = json.loads(decoded_data)
+    access_token = data_json["access_token"]
+    # print(access_token)
+    # print(f"The type of access_token is: {type(access_token)}")
+
+    return access_token
 
-    response = requests.post(url, headers=headers, data=data)
-    response.raise_for_status()
-    return response.json()["access_token"]
 
 def get_enphase_data(enphase_system_id: str) -> float:
-    """
+    """ 
     Get live PV generation data from Enphase API v4
-
     :param enphase_system_id: System ID for Enphase API
     :return: Live PV generation in Watt-hours, assumes to be a floating-point number
     """
-    url = f'https://api.enphaseenergy.com/api/v4/{enphase_system_id}/summary'
+    api_key = os.getenv('ENPHASE_API_KEY')
+    access_token = get_enphase_access_token()
+    # print("ACCESS TOKEN: ", access_token)
```

Comment:
Might be worth removing some of the print comments and others you had been using during debugging